### PR TITLE
Add aggregating job results to testgrid

### PR DESCRIPTION
**Pull request by xueqzhan**
Aggregating jobs are added to blocking dashboard of the corresponding release.
Release is determined by the aggregated job prow config.

### CODE REVIEW

**Comment by petr-muller** [comment]
/label tide/merge-method-squash

LGTM (I'm not familiar with the code but I'm as good reviewer as any)  but has conflicts

File: pkg/release/config/config.go
Comment (hongkailiu) [inline pkg/release/config/config.go:37]: Do we have an example of this in https://github.com/openshift/release/tree/master/core-services/release-controller/_releases?

File: pkg/release/config/config.go
Comment (hongkailiu) [inline pkg/release/config/config.go:45]: Changing it to a pointer might break our existing tool.
/cc @droslean

**Review by hongkailiu — COMMENTED** [review]
Not familiar with the `generator`.
Just checked on the change on the RC's configuration.

File: pkg/release/config/config.go
Comment (hongkailiu) [reply pkg/release/config/config.go:45]: I am surprised that unit tests passed.

File: pkg/release/config/config.go
Comment (xueqzhan) [reply pkg/release/config/config.go:37]: Not at this moment. The codes support it. So if anyone adds the config, this should be support by my change.

File: pkg/release/config/config.go
Comment (xueqzhan) [reply pkg/release/config/config.go:45]: I added the check in the client code for nil pointer. Is there anywhere else we are using this?

File: pkg/release/config/config.go
Comment (hongkailiu) [reply pkg/release/config/config.go:37]: >Not at this moment. The codes support it.

Do you mean the code of release controllers? Could you please link?

Comment (xueqzhan) [reply pkg/release/config/config.go:37]: Yes, release controller. Here is the line: https://github.com/openshift/release-controller/blob/e6e5923afde7a61bd763c3f1b3e0e3bef0ab4e9a/cmd/release-controller/sync_verify_prow.go#L73

File: pkg/release/config/config.go
Comment (hongkailiu) [reply pkg/release/config/config.go:45]: At least, this one.
https://github.com/openshift/ci-tools/blob/98048e212a80054c036fdf38a4a31ddce9c5ad9a/pkg/release/config/client.go#L74

Comment (xueqzhan) [reply pkg/release/config/config.go:45]: This one has been addressed. Maybe you are not seeing all my changes? Yes, I did miss that one in my very first commit. :)

Comment (hongkailiu) [reply pkg/release/config/config.go:45]: ohoh. I see.
All good to me for the config's change.

**Comment by xueqzhan** [comment]
@petr-muller Need a lgtm from you again after the merge. Thanks.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**
